.github/workflows/python-publish.yml

```diff
@@ -0,0 +1,39 @@
+# This workflow will upload a Python Package using Twine when a release is created
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#publishing-to-package-registries
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Upload Python Package
+
+on:
+  release:
```

Comment:
Do you want to also do a testpypi release on tags?

Reply:
Doesn't seem necessary?

Reply:
I'd only do it so you can save yourself a bugfix push, it's fine if you don't mess with packaging too often though.

I would however recommend that you do pull down the package and run tests post deploy though - that way you don't have to think about "will it work".

I wrote a re-usable action for this here: https://github.com/MDAnalysis/pypi-deployment/tree/main

You can see an example of it here: https://github.com/MDAnalysis/MDAnalysisData/blob/master/.github/workflows/deploy.yaml

(you can remove the test deployment if you don't want it).

Advantage of the re-usable action is that you only have to update a single place and everything will keep working.